### ISSUE TITLE
BOT connection to the PostgreSQL database

### ISSUE THREAD

**Issue by ErtonDev** [ErtonDev/add-web#8]
# It's time to connect our bot to the database
When I programmed the bot I thought of using `.txt` files as a database. This has worked fine but now we need to have this information online. **This is why we must change all the code related to the** `.txt` **files and implement the API where it is needed.**

# What to know
We have already programmed different scripts to make this easier, this is a closed issue that may be useful:
> #5 - *Automatize database update process*

There is some data that I would like to keep as `.txt` files `(multas.txt, and other files related to the casino/ games)`, if you would like a more detailed description leave a comment.

The code that we should change is the one related to `bolsa/` and `profile/`:
```
file = io.open("path/file.txt", "r") # r to read, w to write, a to append
file_content = file.readlines()
file.close()
```
This code got the lines of `file.txt`:
```
print(file_content[0])
>>> line_1
print(file_content[1])
>>> line_2
...
print(file_content[i])
>>> line_i+1
```
Basically the bot wrote or updated data in files to then read that same information.

---

You will also have to change the code where the bot creates new profiles:
```
create_profile = os.makedirs(f"profile/{ctx.author.id}_profile", exists_ok = True)
```
*You can look at this in line 899 (29/11/2021 9:50, 11/29/2021 9:50)*

# How to do it
The new code should use our API located inside the `web/` folder. This was programmed by @MarcBen05 
Import it as a module:
```
from web.psqlapi import *
```

Then you can start rewriting the code! :)
**LEAVE A COMMENT IF YOU HAVE ANY QUESTION**

**Comment by ErtonDev** [comment]
# Tasks:
bot1.py - @MarcBen05 
mercado.py - @xaviii03 
niveles.py - @Superjavicon

**Comment by Superjavicon** [comment]
niveles.py successfully connected to database

**Comment by ErtonDev** [comment]
Same for mercado.py @xaviii03

**Comment by ErtonDev** [comment]
The bot is working fine for the time being and it is able to GET data from the DB. We haven't been able to use PUT properly though, the API doesn't seem to work properly. We haven't tried with the POST method yet but there could be a problem there too. **Right now fixing the API is our main objective.**

**Comment by ErtonDev** [comment]
# More information about the error
The error description is something like 'Unable to fetch data'

**Comment by ErtonDev** [comment]
Useful sites:
https://www.postgresqltutorial.com/postgresql-python/
https://www.postgresqltutorial.com/postgresql-python/update/

**Comment by ErtonDev** [comment]
# It is finally working but we have some bot errors, we are testing the script:
These are some of the issues we have found:

 - Using the word "cr" instead of "user_cr" for USER table interaction. 🟢
 - Arguments do not work properly in .banco related commands. Look for compra/vende commands. 🟢   
 - Buying won't take you money. 🟢
 - Stock price gets its value from buyer's capital. 🟢
 - When using `.banco nivel` it will say that the user doesn't have an account. 🟢
 - When using `.banco mercado_negro` it will say that the user doesn't have an account. 🟢
 - When using `.banco nivel mejora` it will say the user doesn't have an account. 🟢
 - When asking to see someone's profile, it gives an error. 🟢
 - Excepts shouldn't be using FileNotFoundError anymore. 🟢
 - When using `.banco transac': everything works but there is no actual transaction xd 🟢 

#### Note:
There are some lines of code we could get rid of. Here is an example of what I'm talking about:
```
variable_name = "variable"
different_name = variable_name
```

For the time being, I will put here the run register from the previous run. (Try GET method externally).
```
----- Bot 7.8.1 online -----
#perfil call by xaviii03 = Se muestra cuenta propia: Me
#banco mercado call by xaviii03 = 35 | 28 | 70 | 63
#banco mercado call by xaviii03 = 45 | 31 | 67 | 64
#perfil call by 3rt0n = Se muestra cuenta propia: Me
Ignoring exception in command contador:
Traceback (most recent call last):
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\site-packages\discord\ext\commands\core.py", line 85
 in wrapped
    ret = await coro(*args, **kwargs)
  File "C:\Users\alumne\Desktop\pythonProjectsE\add-web-copia\bot1.py", line 737, in contador
    read_chupapitos = io.open('counter_chupapitos.txt', 'r')
FileNotFoundError: [Errno 2] No such file or directory: 'counter_chupapitos.txt'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\site-packages\discord\ext\commands\bot.py", line 939
 in invoke
    await ctx.command.invoke(ctx)
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\site-packages\discord\ext\commands\core.py", line 86
, in invoke
    await injected(*ctx.args, **ctx.kwargs)
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\site-packages\discord\ext\commands\core.py", line 94
 in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: FileNotFoundError: [Errno 2] No such file
r directory: 'counter_chupapitos.txt'
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
#perfil call by ??????????? = Se muestra cuenta propia: Me
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
#perfil call by ??????????? = Se muestra cuenta propia: Me
ERROR: Failed to get USER data!
ERROR INFO: list index out of range
EXCEPTION TYPE: <class 'IndexError'>
-------------------
#registro call by ??????????? ! AccountExists
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
Ignoring exception in command None:
discord.ext.commands.errors.CommandNotFound: Command "." is not found
#perfil call by 3rt0n = Se muestra cuenta propia: Me
#banco mercado call by 3rt0n = 88 | 24 | 66 | 65
#banco auxilio call by 3rt0n
#banco mercado_negro call by 3rt0n ! AccountNotFoundError
#banco mercado call by Marth Sgar = 91 | 25 | 54 | 77
#banco call by Marth Sgar
#banco mercado_negro call by xaviii03 ! AccountNotFoundError
#perfil call by Marth Sgar = Se muestra cuenta propia: Me
ERROR: Failed to get USER data!
ERROR INFO: column "cr" does not exist
LINE 1: SELECT cr FROM users WHERE user_id = 548955616153174017
               ^

EXCEPTION TYPE: <class 'psycopg2.errors.UndefinedColumn'>
-------------------
#banco compra 2 1 call by 3rt0n ! ArgumentNotFoundError
Traceback (most recent call last):
  File "C:\Users\alumne\Desktop\pythonProjectsE\add-web-copia\mercado.py", line 44, in <module>
    time.sleep(30)
KeyboardInterrupt
Exception ignored in: <function _ProactorBasePipeTransport.__del__ at 0x000001DC5A08D4C0>
Traceback (most recent call last):
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\asyncio\proactor_events.py", line 116, in __del__
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\asyncio\proactor_events.py", line 108, in close
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 746, in call_soon
  File "C:\Users\alumne\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 510, in _check_closed
RuntimeError: Event loop is closed
```

**Comment by ErtonDev** [comment]
Almost done!

**Comment by ErtonDev** [comment]
It looks like we've fixed every bug, though we'll be trying every command to ensure this.

**Comment by ErtonDev** [comment]
Done!
